src/state/opamSysInteract.ml

```diff
@@ -320,28 +323,28 @@ let install_packages_commands s_packages =
     let epel_release = "epel-release" in
     let install_epel =
       if List.mem epel_release packages then
-        ["yum"::"install"::[epel_release]]
+        ["yum"::"install"::yes ["-y"] [epel_release]]
       else []
     in
     install_epel @
-    ["yum"::"install"::
+    ["yum"::"install"::yes ["-y"]
      (OpamSysPkg.Set.remove (OpamSysPkg.of_string epel_release) s_packages
       |> OpamSysPkg.Set.elements
       |> List.map OpamSysPkg.to_string);
      "rpm"::"-q"::"--whatprovides"::packages]
   | "bsd" ->
     if spv OpamSysPoll.os_distribution = "freebsd" then
-      ["pkg"::"install"::packages]
+      ["pkg"::"install"::yes ["-y"] packages]
     else
-      ["pkg_add"::packages]
+      ["pkg_add"::yes ~no:["-I"] ["-i"] packages]
```

Comment:
This is correct for openbsd.  I suspect we need to distinguish `netbsd` and `openbsd` here, since their package managers are a different implementation now.

On OpenBSD, you can also add
```
     -a       Automated package installation; do not record packages as
              installed manually.
```

which marks it as an automatically installed package for the purposes of system cleanup.

Reply:
Wouldn't that make any clean-up of the packages automatically remove it?

Reply:
Not on OpenBSD -- it's just used to give a cleaner output in `pkg_info` (to only show packages you've chosen to install by default, and not their dependencies).

> pkg_add differentiates between packages specified on the command line, and packages installed automatically because of inter-dependencies: the first kind will be tagged as ‘installed manually’. The -a option is used internally by the ports(7) infrastructure and dpb(1) to handle dependencies.
>
> It is also possible to tweak the ‘installed manually’ status of a package after the fact. Running pkg_add on an already installed package will tag it as ‘installed manually’, even if it was already there as a dependency of something else, and doubling the -a option will remove the ‘installed manually’ tag from installed packages.
>
> pkg_info(1) can be used to show only manually-installed packages, and pkg_delete(1) can be used to remove dependencies when they are no longer needed.

Reply:
That's what I mean with the last bit. How does pkg_delete know it's no longer needed, given that ports has no knowledge of opam's dependency on it?

Reply:
Or is the point that `pkg_delete` never does that automatically, the user determines it?

Reply:
The user determines it. It's only ever manually done by `pkg_delete -a`

Reply:
I've no idea about what `opam depext` semantics should be in this respect, but note that FreeBSD has a similar `-A` argument which could be passed.

Reply:
About marking packages as automatically installed, we thought about proposing that  on the opam package removal, like that the system package is no more marked as manually installe din its system. But it can lead to a removal of the system package on the next cleanup, and if a package that depends on is installed on another switch, it can lead to a reinstall.